File: analyses/post_analysis/combine_all_hits.R
library(tidyverse)
library(arrow)

exp_dir <- "/Volumes/bluecub/aglycone_release_100um_24h/output"
res_out <- "/Volumes/bluecub/aglycone_release_100um_24h/combined_results"

exp_names <- list.files(
  exp_dir,
  recursive = FALSE,
  full.names = FALSE
)

all_hits <- exp_names %>%
  purrr::map(
    .x = .,
    .f = ~ {
      bio <- file.path(
        exp_dir,
        .x,
        "report",
        "biotransformer_similarities.parquet"
      )

      anno <- file.path(
        exp_dir,
        .x,
        "report",
        "annotation_similarities.parquet"
      )

      anno_wrangle <- arrow::read_parquet(file = anno) %>%
        dplyr::mutate(
          experiment = .x,
          source = "anno",
          sim_mol = NA
        ) %>%
        dplyr::select(
          "experiment",
          "source",
          "sim_mol",
          "feature" = peak_id,
          "feature_pred" = feature,
          "adduct",
          "mass" = "exact_mass",
          "mz",
          "rtime",
          "sim",
          "formula" = "target_formula",
          "smiles" = "target_smiles",
          "inchikey" = "target_inchikey",
          "ppm_error"
        )

      bio_wrangle <- arrow::read_parquet(file = bio) %>%
        dplyr::mutate(
          experiment = .x,
          source = "bio",
          feature_pred = paste0(feature, ".", met_id),
          ppm_error = abs((mzmed - mz) / mz * 1e6)
        ) %>%
        dplyr::select(
          "experiment",
          "source",
          "sim_mol",
          "feature",
          "feature_pred" = feature_pred,
          "adduct",
          "mass",
          "mz" = "mzmed",
          "rtime" = "rtmed",
          "sim",
          "formula" = `Molecular formula`,
          "smiles" = "SMILES",
          "inchikey",
          "ppm_error"
        )

      dplyr::bind_rows(
        ... = list(
          anno_wrangle,
          bio_wrangle
        )
      )
    }
  ) %>%
  dplyr::bind_rows()

arrow::write_parquet(
  x = all_hits,
  sink = file.path(
    res_out,
    "aglycone_release_100um_24h_hits.parquet"
  )
)

File: analyses/post_analysis/combine_all_hits.py
# %%
import os
from pathlib import Path
import polars as pl

work_dir = Path(
    f"/Users/wilhelm/Proton/01_juniper/01_arbete/01_projekt/03_psm/"
    f"metabolite_disco"
)
exp_dir = Path("/Volumes/bluecub/aglycone_release_100um_24h/output")
out_path = Path("/Volumes/bluecub/aglycone_release_100um_24h/combined_results")
out_save = Path(out_path, "aglycone_release_100um_24h_hits.parquet")

exp_names = (
    [i for i in os.listdir(exp_dir) if os.path.isdir(os.path.join(exp_dir, i))]
)

exp_names.sort()

exp_path = Path(
    exp_dir,
    exp_names[0],
    "report",
    "similar_hits.parquet"
)

data = pl.read_parquet(exp_path)

all_hits = pl.concat([
    pl.read_parquet(Path(exp_dir, name, "report", "similar_hits.parquet")) 
    for name in exp_names
])

all_hits.write_parquet(file=out_save)


# %%
